### PR TITLE
RUM-6286: replacing `joinToString` when it possible, refactor a bit

### PR DESCRIPTION
**Pull request by satween**
### What does this PR do?

Here we're trying to reduce usages of joinToString method inside the project, replacing it with more performant string builder 

### Motivation

Using `joinToString`  may cause a lot of memory and cpu consumption which we trying to avoid, minimizing impact of sdk to user's code 

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        val initialData = "something inside buffer already"
+        val buffer = StringBuilder(initialData)
+
+        // When
+        buffer.appendIfNotEmpty(' ')
+
+        // Then
+
+        assertThat(buffer.toString()).isEqualTo("$initialData ")
+    }
```
Comment (xgouchet) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:20]: > [!WARNING]
> When you test this, even though the implementation is trivial, you're only ever testing that it works for `"something inside buffer already"`. 
> Instead you might want to use a property based approach : create a random non-empty string (we use the regex feature to ensure that there's at least one character, of any kind) and use that as input.

```suggestion
        @Test
    fun `M add char W addIfNotEmpty {buffer is not empty}`(
        @StringForgery(regex = ".+") initialContent: String
    ) {
        // Given
        val buffer = StringBuilder(initialContent)

        // When
        buffer.appendIfNotEmpty(' ')

        // Then
        assertThat(buffer.toString()).isEqualTo("$initialContent ")
    }
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    fun `M add str W addIfNotEmpty {buffer is not empty}`() {
```
Comment (xgouchet) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:23]: > [!NOTE]
> Same as above applies here

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                        itemsForJoin.forEachIndexed { i, item ->
+                            if (i > 0) append(" ")
+                            append(item)
+                        }
+                    }
+                }
+            )
+        }
+
+        assertThat(builderResult).isEqualTo(jointToStringResult) // same result
```
Comment (xgouchet) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt:61]: > [!WARNING]
> You're only making that assertion on the last iteration of your test. You could move this inside the loop to ensure every iteration behaves correctly.

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    ExtendWith(ForgeExtension::class)
+)
+@ForgeConfiguration(Configurator::class)
+internal class PerformanceTest {
+
+    private fun List<Long>.mean() = sum().toDouble() / size
+
+    @Test
+    fun `M be faster than joinToString W buildString`(forge: Forge) {
+        val itemsForJoin = forge.aList(ITEMS_TO_JOINT) { forge.aString() }
```
Comment (xgouchet) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt:31]: > [!NOTE]
> You're running all the iterations on the same dataset. On the one hand, it gives a more relevant mean, but I'm curious whether we could get more insights if we moved the `itemsForJoin` inside the loop and have a different data set each time. 
> Also we're looking only at the mean, what about other insights (p90, stdev, etc…), can we learn info about edge cases from those?

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt (hunk 1 above)
Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt]: good suggestion, applied

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt (hunk 2 above)
Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt]: 👍

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt (hunk 3 above)
Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt]: done, thanks

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt (hunk 4 above)
Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt]: Moving `itemsForJoin` creation inside the loop will significantly increase test execution time, so I left it outside for now. 

Regarding the p95 - it's a good point. After some research I've noticed that the STD value is relatively big, so mean is not the best metric for this measurement. I've changed  metric to p95 as it seems good compromise between accuracy and sustainability to flakiness. Still test is highly depends on the machine state, so if we will notice that it flaky will reduce metric to p90

Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/PerformanceTest.kt]: Upd: reduced to p90 for more stability

**Review by xgouchet — APPROVED** [review]

File: dd-sdk-android-core/api/apiSurface [hunk 5]
```diff
@@ -190,6 +190,8 @@ class com.datadog.android.core.SdkReference
   fun get(): com.datadog.android.api.SdkCore?
 fun <T> allowThreadDiskReads(() -> T): T
 fun <T> allowThreadDiskWrites(() -> T): T
+fun StringBuilder.appendIfNotEmpty(String)
+fun StringBuilder.appendIfNotEmpty(Char)
```
Comment (0xnm) [inline dd-sdk-android-core/api/apiSurface:194]: These functions should be in the `dd-sdk-android-internal` module instead. The problem here is that `core` module is re-exported to the client, so all the symbols here are visible to the client, but we don't want to pollute (make these symbols to appear in the suggestions dropdown) namespace of methods for classes from stdlib, this is only for our usage.

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core
+
+/**
+ * This utility function helps replace joinToString calls with more efficient StringBuilder's methods calls.
+ * In case when content should be separated with some separator it's handy to add it in front of new string only
+ * if buffer already contain some data
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt:12]:
```suggestion
 * In case when content should be separated with some separator, it's handy to add it in front of a new string only
 * if buffer already contains some data.
```

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ * @param str - string that should be added into buffer only if it already contains some data
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt:13]:
```suggestion
 * @param str string that should be added to the buffer only if it already contains some data.
```

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+fun StringBuilder.appendIfNotEmpty(str: String) = apply {
+    if (isNotEmpty()) append(str)
+}
+
+/**
+ * This utility function helps replace joinToString calls with more efficient StringBuilder's methods calls.
+ * In case when content should be separated with some separator it's handy to add it in front of new string only
+ * if buffer already contain some data
+ * @param char - char that should be added into buffer only if it already contains some data
+ */
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt:24]:
```suggestion
/**
 * This utility function helps to replace [joinToString] calls with more efficient [StringBuilder] methods calls.
 * In case when content should be separated with some separator, it's handy to add it in front of a new string only
 * if buffer already contains some data.
 * @param char char that should be added to the buffer only if it already contains some data.
 */
```

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/error/internal/DatadogExceptionHandler.kt [hunk 9]
```diff
@@ -133,12 +133,11 @@ internal class DatadogExceptionHandler(
...
             .filterKeys { it != crashedThread }
             .filterValues { it.isNotEmpty() }
-            .map {
-                val thread = it.key
+            .map { (thread, stackTrace) ->
                 ThreadDump(
                     name = thread.name,
                     state = thread.state.asString(),
-                    stack = it.value.loggableStackTrace(),
+                    stack = stackTrace.loggableStackTrace(),
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/error/internal/DatadogExceptionHandler.kt:140]: seems not related with the change done in the PR? just a small side improvement?

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.jupiter.api.extension.Extensions
+
+@Extensions(
+    ExtendWith(ForgeExtension::class)
+)
+internal class StringBuilderExtKtTest {
+
+    @Test
+    fun `M add char W addIfNotEmpty {buffer is not empty}`(
+        @StringForgery(regex = ".+") initialContent: String
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:17]: probably regex is not needed here and any string is good? In this case you can leave only `@StringForgery`, without explicit argument? it will always generate non-empty string

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+package com.datadog.android.core
+
+/**
+ * This utility function helps replace joinToString calls with more efficient StringBuilder's methods calls.
+ * In case when content should be separated with some separator it's handy to add it in front of new string only
+ * if buffer already contain some data
+ * @param str - string that should be added into buffer only if it already contains some data
+ */
+fun StringBuilder.appendIfNotEmpty(str: String) = apply {
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt:15]: side thought: naming can be a bit confusing, because without looking at the method description I initially thought that it checks argument for being empty, not receiver.

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    ) {
+        // Given
+        val buffer = StringBuilder(initialContent)
+
+        // When
+        buffer.appendIfNotEmpty(' ')
+
+        // Then
+
+        assertThat(buffer.toString()).isEqualTo("$initialContent ")
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:27]:
```suggestion
        // Then
        assertThat(buffer.toString()).isEqualTo("$initialContent ")
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Test
+    fun `M add str W addIfNotEmpty {buffer is not empty}`(
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:31]:
```suggestion
    fun `M add string W addIfNotEmpty {buffer is not empty}`(
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+    ) {
+        // Given
+        val buffer = StringBuilder(initialContent)
+
+        // When
+        buffer.appendIfNotEmpty(" ")
+
+        // Then
+
+        assertThat(buffer.toString()).isEqualTo("$initialContent ")
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:42]:
```suggestion
        // Then
        assertThat(buffer.toString()).isEqualTo("$initialContent ")
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Test
+    fun `M add char W addIfNotEmpty {buffer is not empty}`(
+        @StringForgery(regex = ".+") initialContent: String
+    ) {
+        // Given
+        val buffer = StringBuilder(initialContent)
+
+        // When
+        buffer.appendIfNotEmpty(' ')
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:23]: nit: ideally you should append a random char, not test it only with `' '` (same for other tests).

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        // When
+        buffer.appendIfNotEmpty(" ")
+
+        // Then
+
+        assertThat(buffer.toString()).isEqualTo("$initialContent ")
+    }
+
+    @Test
+    fun `M not add any char W addIfNotEmpty {buffer is empty}`() {
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:46]: nit: test name contradicts with test execution - it is always `' '` char added here, not any.

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+        // Given
+        val buffer = StringBuilder()
+
+        // When
+        buffer.appendIfNotEmpty(' ')
+
+        // Then
+
+        assertThat(buffer.toString()).isEqualTo("")
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:55]:
```suggestion
        // Then
        assertThat(buffer.toString()).isEmpty()
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+    fun `M not add any str W addIfNotEmpty {buffer is empty}`() {
+        // Given
+        val buffer = StringBuilder()
+
+        // When
+        buffer.appendIfNotEmpty(" ")
+
+        // Then
+
+        assertThat(buffer.toString()).isEqualTo("")
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt:68]:
```suggestion
        // Then
        assertThat(buffer.toString()).isEmpty()
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+import kotlin.math.sqrt
+import kotlin.system.measureNanoTime
+
+private const val ITEMS_TO_JOINT = 10_000
+private const val REPETITION_COUNT = 10_000
+
+@Extensions(
+    ExtendWith(ForgeExtension::class)
+)
+@ForgeConfiguration(value = Configurator::class, seed = 0x50c72968d123L)
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt:28]: seed has to be removed, probably it is left after the debugging

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import kotlin.math.pow
+import kotlin.math.round
+import kotlin.math.sqrt
+import kotlin.system.measureNanoTime
+
+private const val ITEMS_TO_JOINT = 10_000
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt:22]:
```suggestion
private const val ITEMS_TO_JOIN = 10_000
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+private const val REPETITION_COUNT = 10_000
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt:23]: we usually wrap constants in the test in the nested companion object at the bottom of the test class, so that everyone knows what is the layout.

File: detekt_custom.yml [hunk 22]
```diff
@@ -1245,6 +1245,7 @@ datadog:
       - "kotlin.text.StringBuilder()"
       - "kotlin.text.buildString(kotlin.Function1)"
       - "kotlin.text.buildString(kotlin.Int, kotlin.Function1)"
+      - "java.lang.StringBuilder.clear()"
```
Comment (0xnm) [inline detekt_custom.yml:1248]: this has to be moved to the right position in this file: it doesn't belong to the `Kotlin String` region and once proper region found, the position should be in the alphabetical order.

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+                " mean = ${jointToStringExecutionTime.mean}\n" +
+                " std = ${jointToStringExecutionTime.std}\n" +
+                " cv = ${"%.2f".format(jointToStringExecutionTime.cv)}%\n" +
+                " p50 = ${jointToStringExecutionTime.percentile(50)},\n" +
+                " p90 = ${jointToStringExecutionTime.percentile(90)},\n" +
+                " p95 = ${jointToStringExecutionTime.percentile(95)},\n" +
+                " p99 = ${jointToStringExecutionTime.percentile(99)}\n"
+            )
+
+        println(statisticsReport)
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt:82]: debugging leftover?

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+@Extensions(
+    ExtendWith(ForgeExtension::class)
+)
+@ForgeConfiguration(value = Configurator::class, seed = 0x50c72968d123L)
+internal class JointToStringVsStringBuilderPerformanceTest {
+
+    @Test
+    fun `M be faster than joinToString W buildString`(forge: Forge) {
+        val itemsForJoin = forge.aList(ITEMS_TO_JOINT) { forge.aString() }
+        val jointToStringExecutionTime = mutableListOf<Long>()
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt:34]:
```suggestion
        val joinToStringExecutionTime = mutableListOf<Long>()
```

File: dd-sdk-android-core/api/apiSurface (hunk 5 above)
Comment (satween) [reply dd-sdk-android-core/api/apiSurface]: moved, thanks for clarifications

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt (hunk 6 above)
Comment (satween) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt]: 👍

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt (hunk 7 above)
Comment (satween) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt]: 👍

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt (hunk 8 above)
Comment (satween) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt]: 👍

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt (hunk 20 above)
Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt]: 👍

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt (hunk 21 above)
Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt]: done, will do next time right way

File: detekt_custom.yml (hunk 22 above)
Comment (satween) [reply detekt_custom.yml]: I thought that if it's extension function it should be in kotlin space, anyway, moved

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt (hunk 23 above)
Comment (satween) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/JointToStringVsStringBuilderPerformanceTest.kt]: The idea was to print this information, but without failed test nobody will read it. Changed to `withFailMessage` instruction

File: detekt_custom.yml (hunk 22 above)
Comment (0xnm) [reply detekt_custom.yml]: Then it should be `kotlin.text.StringBuilder.clear()` here.

Comment (satween) [reply detekt_custom.yml]: `java.lang.StringBuilder.clear()` is how detekt sees it. Probably there is some issue in `UnsafeThirdPartyFunctionCall` but adding `kotlin.text.StringBuilder.clear()` to whitelist will not help.

File: dd-sdk-android-internal/src/test/java/com/datadog/internal/utils/StringBuilderExtKtTest.kt [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+
+@Extensions(
+    ExtendWith(ForgeExtension::class)
+)
+internal class StringBuilderExtKtTest {
+
+    @Test
+    fun `M add char W addIfNotEmpty {buffer is not empty}`(
```
Comment (0xnm) [inline dd-sdk-android-internal/src/test/java/com/datadog/internal/utils/StringBuilderExtKtTest.kt:23]: nit: the method name is actually a bit different (same for other tests in this file)

```suggestion
    fun `M add char W appendIfNotEmpty {buffer is not empty}`(
```

File: dd-sdk-android-internal/src/test/java/com/datadog/internal/utils/StringBuilderExtKtTest.kt [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+    ) {
+        // Given
+        val buffer = StringBuilder(initialContent)
+
+        // When
+        buffer.appendIfNotEmpty(' ')
+
+        // Then
+
+        assertThat(buffer.toString()).isEqualTo("$initialContent ")
```
Comment (0xnm) [inline dd-sdk-android-internal/src/test/java/com/datadog/internal/utils/StringBuilderExtKtTest.kt:34]:
```suggestion
        // Then
        assertThat(buffer.toString()).isEqualTo("$initialContent ")
```

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt (hunk 11 above)
Comment (satween) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/StringBuilderExt.kt]: decided to keep as is

**Review by 0xnm — APPROVED** [review]
